VimeoUpload/Upload/Controllers/VideoDeletionManager.swift

```diff
@@ -106,14 +126,19 @@ public class VideoDeletionManager: NSObject
             }
         }
         
-        return KeyedArchiver(basePath: deletionsArchiveDirectory.path)
+        return KeyedArchiver(basePath: deletionsArchiveDirectory.path, archivePrefix: archivePrefix)
     }
     
     // MARK: Archiving
     
     private func loadDeletions() -> [VideoUri: Int]
     {
-        if let deletions = self.archiver?.loadObject(for: type(of: self).DeletionsArchiveKey) as? [VideoUri: Int]
+        guard self.shouldLoadArchive == true else
+        {
+            return [:]
```

Comment:
Would it be worth it to just return nil here and change the method signature to return an optional?